sensing/src/main/java/com/google/android/sensing/upload/UploadRequestFetcher.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.sensing.upload
+
+import com.google.android.sensing.SensingEngine
+import com.google.android.sensing.model.RequestStatus
+import com.google.android.sensing.model.UploadRequest
+
+class UploadRequestFetcher(private val sensingEngine: SensingEngine) {
+  suspend fun fetch(): List<UploadRequest> {
```

Comment:
Can you rename the method name to `fetchPendingRequests()` or `fetch(var uploadRequestStates)`? Alternatively, you could also write a private method in `SensingSynchornizer` which fetches this data as this class is not doing much.

Reply:
Changed it to fetchForUpload()

Want to keep this logic separate from `SensingSynchronizer` so that there is no direct dependency on `SensingEngine` .